examples/Microsoft.CognitiveServices_accounts_connections@2025-06-01/main.tf

```diff
@@ -0,0 +1,307 @@
+  type      = "Microsoft.CognitiveServices/accounts/connections@2025-06-01"
+  parent_id = azapi_resource.account.id
+  name      = "${var.resource_name}-apikey"
+  body = {
+    properties = {
+      authType = "ApiKey"
+      category = "AzureOpenAI"
+      target   = azapi_resource.account_openai.output.properties.endpoint
+      metadata = {
+        ApiType    = "Azure"
+        ResourceId = azapi_resource.account_openai.id
+        location   = var.location
+      }
+    }
+  }
+  sensitive_body = {
+    properties = {
+      credentials = {
+        key = azapi_resource_action.account_openai_keys.sensitive_output.key1
+      }
+    }
+  }
+  schema_validation_enabled = false
+  ignore_casing             = false
+  ignore_missing_property   = false
+  depends_on                = [azapi_resource.connection_aad]
+}
+
+resource "azapi_resource" "connection_customkeys" {
+  type      = "Microsoft.CognitiveServices/accounts/connections@2025-06-01"
+  parent_id = azapi_resource.account.id
+  name      = "${var.resource_name}-custom"
+  body = {
+    properties = {
+      authType = "CustomKeys"
+      category = "CustomKeys"
+      target   = azapi_resource.account_openai.output.properties.endpoint
+      metadata = {
+        ApiType    = "Azure"
+        ResourceId = azapi_resource.account_openai.id
+        location   = var.location
+      }
+    }
+  }
+  sensitive_body = {
+    properties = {
+      credentials = {
+        keys = {
+          primaryKey   = azapi_resource_action.account_openai_keys.sensitive_output.key1
+          secondaryKey = azapi_resource_action.account_openai_keys.sensitive_output.key2
+        }
+      }
+    }
+  }
+  schema_validation_enabled = false
+  ignore_casing             = false
+  ignore_missing_property   = false
+  depends_on                = [azapi_resource.connection_apikey]
+}
+
+# This is example is based on having an external resource that uses OAuth2. 
+resource "azapi_resource" "connection_oauth" {
+  type      = "Microsoft.CognitiveServices/accounts/connections@2025-06-01"
+  parent_id = azapi_resource.account.id
+  name      = "${var.resource_name}-oauth"
+  body = {
+    properties = {
+      authType = "OAuth2"
+      category = "AzureBlob"
+      target   = azapi_resource.storageAccount.output.properties.primaryEndpoints.blob
+      metadata = {
+        containerName = azapi_resource.container.name
+        accountName   = azapi_resource.storageAccount.name
+      }
+    }
+  }
+  sensitive_body = {
+    properties = {
+      credentials = {
+        # Not all fields are required.
+        # Use the fields that are necessary in an actual use of the credentials, you don't need to use all of them, they are just placeholders for validation in this connection.
+        authUrl        = "https://login.microsoftonline.com/tenantplaceholder/oauth2/v2.0/token"
+        clientId       = "placeholderClientId"
+        clientSecret   = "placeHolderClientSecret"
```

Comment:
For the secrets, please use sensitive variables.

Reply:
fixed